learnosity_sdk/request/dataapi.py

```diff
@@ -7,8 +9,8 @@
 
 class DataApi(object):
 
-    def request(self, endpoint, security_packet,
-                secret, request_packet={}, action='get'):
+    def request(self, endpoint: str, security_packet: Dict[str, str],
+                secret: str, request_packet:Dict[str, Any] = {}, action: str = 'get') -> Response:
```

Comment:
Codacy has a point. It's in pre-existing code, but it might be worth making the `request_packet` `Optional`, and transform `None` to `{}` in the code if needed.

FWIW, when you have a mutable object as a default argument, this object remains for the duration of your run.

This can lead to... confusing behaviours.
![Screenshot 2024-08-20 at 06 50 02](https://github.com/user-attachments/assets/1ad8b443-e437-454a-8b66-945640534d2f)

Reply:
There are a few places where the logic might need to be tweaked to address this Codacy issue. I don't mind creating a ticket and actioning this in a separate PR to avoid increasing the impact of this PR